### ISSUE TITLE
Colour contrast for disabled button doesn't meet WCAG 2.1

### ISSUE THREAD

**Issue by maspychaj-sage** [Sage/carbon#4346]
### Current behaviour

![image](https://user-images.githubusercontent.com/75613789/131462417-a8a6e7f9-18de-4056-9eae-f909df64f4fc.png)

### Expected behaviour

Contrast of disabled buttons should meet requirements of WCAG 2.1

### CodeSandbox or Storybook URL

https://codesandbox.io/s/eloquent-galois-sfujx?file=/src/index.js

### Carbon Version

83.0.2

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

It occurs on any browser and operating system.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by nicktitchmarsh** [comment]
@tempertemper is this something UX needs to address? We can change the theme colours to make this more accessible

**Comment by tempertemper** [comment]
@nicktitchmarsh Yeah, it's a design thing to work out the right solution so that this passes the AA colour contrast threshold. So it needs to go to UX and then, once we have a solution, up to the Design System team for the 'ok' before it gets to Carbon.

For me, disabled buttons cause more problems than they solve, so I'd take them out of the DS entirely. But I'm not sure how practical that would be 😆

**Comment by nicktitchmarsh** [comment]
@harpalsingh Please could UX look at this issue and raise a ticket in Carbon to fix once you have a solution?

**Comment by tempertemper** [comment]
.@harpalsingh is right: it's not a fail by the letter of the WCAG law as it's an inactive control, but for me this is a doing-right-by-our-users thing. Primarily for users with low vision, who really benefit from sufficient contrast. They get > 4.5:1 contrast on the rest of the page, most of which can't be interacted with, so why not disabled buttons?

That's easier said than done though. This is a Design System issue, rather than directly Carbon, and some proper design time needs to be allocated. The issue is that [increasing the contrast of the text](https://webaim.org/resources/contrastchecker/?fcolor=69696D&bcolor=E6EBED) is all well and good, but a significant proportion of users may then see the button as clickable. User testing and design iterations will reassure us that we're going in the right direction.

The first set of buttons in the following image are the current design, showing a failing colour contrast between the button background and the button text; the second set of buttons pass the contrast threshold:

<img width="362" alt="Three sizes of buttons that do not pass colour contrast, then three that do" src="https://user-images.githubusercontent.com/2418781/139857155-e7ee4af4-ead0-442e-83ff-4b0d9d0e3113.png">

We also need to think about other disabled components (form inputs, for example) that might need to change appearance in order to give a consistent visual experience to users.

Depending on the design we land on, the roll-out might need to be carefully managed. I suspect in a lot of cases, removing the disabled button and using some smart content or carefully designed interaction patterns would be a better solution. But, again, easier said than done.

For me, I'd ban disabled UI elements entirely as they cause more trouble than they're worth 😂 But I know there's a valid UI-of-last-resort use-case, so I won't be pushing for that!

---

On the issue that this Issue stemmed from, I thought to run some tests on the Codesandbox and the contrast of the button isn't picked up by axe. This is likely because the button has the `disabled` attribute. Replacing the `disabled` attribute with `aria-disabled="true"` also fails to trigger an axe issue.

This would suggest one of two things:

1. The products where this issue is being raised by axe aren't adding the `disabled` attribute and are styling the button to *look* disabled and changing the behaviour
2. The component didn't have the `disabled` attribute when the issue arose, but does now
3. axe has refined its algorithm between the issue being raised and now, meaning it *was* flagging UI elements with disabled states, but now it's not

The first would be very odd as it would be a lot more work for a developer, rather than using the component's inbuilt 'disabled' feature. One way or the other, it's probably not something we should be doing without some proper design consideration.

In the meantime, if the offending instance could be replaced with a different design, or the implementation could be checked to ensure the `disabled` attribute is being properly added, that would be great.

Longer term, we should look at a design revision for the component. I'll add a ticket to look at this to the design Jira board.

**Comment by v99rly** [comment]
Our advice would be that disabled buttons are not used unless it's clear what action has to be performed to enable them. The contrast issue isn't actually an issue for now as disabled buttons don't actually have to meet 4.5:1 contrast ratios as per the guidance from WCAG, However that doesn't mean that this method is accessible. The Design Team have a ticket in their backlog to have a further discussion about this and come up with a more suitable solution.

**Comment by samtjo** [comment]
Hi @v99rly - can I ask who you are please?

**Comment by nicktitchmarsh** [comment]
@v99rly do you know if this has been discussed yet?

**Comment by ingridzillinger** [comment]
The design system team has discussed this and we support what Martin writes above: Increasing the contrast of disabled buttons can cause confusion as the button looks as if it is clickable. Our plan is to ask the teams for real life examples that include other disabled information (e.g. disabled text inputs) to see how this works together and visually match them.

**Comment by nicktitchmarsh** [comment]
I am going to close this issue in favour of Design System providing guidance on the usage of disabled buttons
